networking/src/p2p/peer.rs

```diff
@@ -582,23 +825,33 @@ async fn begin_process_incoming(
     while net.rx_run.load(Ordering::Acquire) {
         match timeout(READ_TIMEOUT_LONG, rx.read_message::<PeerMessageResponse>()).await {
             Ok(res) => match res {
-                Ok(msg) => {
-                    let should_broadcast_message = net.rx_run.load(Ordering::Acquire);
-                    if should_broadcast_message {
-                        trace!(log, "Message parsed successfully"; "msg" => format!("{:?}", &msg));
-                        event_channel.tell(
-                            Publish {
-                                msg: PeerMessageReceived {
-                                    peer: myself.clone(),
-                                    message: Arc::new(msg),
-                                }
-                                .into(),
-                                topic: NetworkChannelTopic::NetworkEvents.into(),
-                            },
-                            None,
-                        );
+                Ok(msg) => match throttle_quota.lock() {
+                    Ok(ref mut quota) => {
+                        if let Err(quota) = quota.can_receive(&msg) {
+                            if quota == -1 {
```

Comment:
@akoptelov 
I dont like this second if `if quota == -1 {` , it is not that clear, 
if there is a problem with quota, then the first if `if let Err(quota)` should be enought and I would just log exceeeded quota number to log

Reply:
@akoptelov 
ok, so I would do it better reverse:
- the whole IF with log `if quota == -1 {`  just move inside to the `quota.can_receive`
- so here we just need simple if `if quota.can_receive(&msg).is_ok() {`
  -  or maybe easier, just return `bool` on `quota.can_receive`